src/Paystack.php

```diff
@@ -98,59 +98,39 @@ private function setRequestOptions()
 
     /**
      * Initiate a payment request to Paystack
-     * @return Unicodeveloper\Paystack\Paystack
      */
     public function makePaymentRequest()
     {
-        $this->setResponse('/transaction/initialize');
-
-        return $this;
-    }
-
-    /**
-     * Make the client request and get the response
-     * @param string $relativeUrl
-     * @return Unicodeveloper\Paystack\Paystack
-     */
-    public function setResponse($relativeUrl)
-    {
         $data = [
             "amount" => intval(request()->amount),
             "reference" => request()->reference,
             "email" => request()->email
         ];
 
-        $this->response = $this->client->post($this->baseUrl . $relativeUrl, [
-            'body' => json_encode($data)
-        ]);
+        $this->setHttpResponse('/transaction/initialize', 'POST', $data);
 
         return $this;
     }
 
-    private function setGetResponse($relativeUrl, $body = [])
-    {
-        $this->response = $this->client->get($this->baseUrl . $relativeUrl, $body);
 
-        return $this;
-    }
+    private function setHttpResponse($relativeUrl, $method, $body = []){
 
-    private function setPostResponse($relativeUrl, $body = [])
-    {
-        $this->response = $this->client->post($this->baseUrl . $relativeUrl, $body);
+        if(is_null($method)){
 
-        return $this;
-    }
+            throw new isNullException("Empty method not allowed");
 
-    private function setPutResponse($relativeUrl, $body = [])
-    {
-        $this->response = $this->client->put($this->baseUrl . $relativeUrl, $body);
+        }else{
+
+            $this->response = $this->client->{strtolower($method)}($this->baseUrl . $relativeUrl, $body);
+
```

Comment:
Something is still wrong with they way we are implementing this post request. When making a post request we are meant to have this in the array `['body' => json_encode($data)]`. The way it is right now, do we have that?